src/components/generic/Select/Select.jsx

```diff
@@ -46,6 +46,18 @@ export default function Select({
 
   const onOutsideClick = () => setExpanded(false);
 
+  const handleKeyDown = (event) => {
+    if (event.key === 'Enter' || event.key === ' ') {
```

Comment:
maybe you should also check that the event originated in this element, not a child element (an option), because clicking Enter on an option should select that option, which I guess collapses the dropdown anyway, and an additional toggleExpanded would re-open it.
```js
event.target === event.currentTarget